### ISSUE TITLE
Potential future breakage on linux

### ISSUE THREAD

**Issue by hoytech** [hoytech/vmtouch#74]
It looks like mincore() may crippled in new linux kernels, to protect against side-channel attacks: [this linux kernel commit](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=134fca9063ad4851de767d1768180e5dede9a881)

Unfortunately, we may need to run vmtouch as root to get accurate page-count information on files we don't have write access to.

**Comment by freedge** [comment]
indeed, we experienced issue with SLES12 and kernel 4.12.14-95.24 - vmtouch would just report every single file as fully present in the page cache.

It's a bit messy, Linus [committed](https://www.theregister.co.uk/2019/01/08/linux_patch_page_cache/), then reverted, a patch for mincore. Not sure how newer kernel versions will behave.
